### ISSUE TITLE
Train from scratch but got a low accm and acct

### ISSUE THREAD

**Issue by transform19** [FoundationVision/Infinity#65]
Hi Team,

I have been training Infinity from scratch using my own dataset, but I am getting disappointing results – both accm and acct are only reaching 0.70. 

According to other issues, an accuracy of around 80% is considered good. 

Could you please provide some advice on how to train Infinity from scratch more effectively?

**Comment by JeyesHan** [comment]
@transform19 Can you please tell me about your experiment setup, like data volume, number of GPUs, and characteristics of the data?

**Comment by transform19** [comment]
My dataset is a subset of Laion-5B with roughly 5M samples recaptioned using QWen-VL. The official dataset is replaced with WebDataset to fit my dataset format.

Model is trained on 4 GPUs with lbs of 16. 

This is the detail of my experiment setup, most of which followed the script/train.sh. 

`torchrun \
--nproc_per_node=4 \
--nnodes=1 \
--node_rank=${node_rank} \
--master_addr=${master_addr} \
--master_port=${master_port} \
train.py \
--ep=10 \
--opt=adamw \
--cum=3 \
--sche=lin0 \
--fp16=2 \
--ada=0.9_0.97 \
--tini=-1 \
--tclip=5 \
--flash=0 \
--alng=5e-06 \
--saln=1 \
--cos=1 \
--enable_checkpointing=full-block \
--local_out_path ${local_out_path} \
--task_type='t2i' \
--bed=${bed_path} \
--data_path=${data_path} \
--video_data_path=${video_data_path} \
--exp_name=${exp_name} \
--tblr=6e-5 \
--pn 0.06M \
--model=2bc8 \
--lbs=16 \
--workers=8 \
--short_cap_prob 0.5 \
--online_t5=1 \
--use_streaming_dataset 1 \
--iterable_data_buffersize 30000 \
--Ct5=2048 \
--t5_path=.cache/huggingface/hub/ \
--vae_type 32 \
--vae_ckpt=weights/infinity_vae_d32reg.pth  \
--wp 0.00000001 \
--wpe=1 \
--dynamic_resolution_across_gpus 1 \
--enable_dynamic_length_prompt 1 \
--reweight_loss_by_scale 1 \
--add_lvl_embeding_only_first_block 1 \
--rope2d_each_sa_layer 1 \
--rope2d_normalized_by_hw 2 \
--use_fsdp_model_ema 0 \
--always_training_scales 100 \
--use_bit_label 1 \
--zero=2 \
--save_model_iters_freq 2000 \
--log_freq=1 \
--checkpoint_type='torch' \
--prefetch_factor=16 \
--noise_apply_strength=0.3 \
--noise_apply_layers=13 \
--apply_spatial_patchify 0 \
--use_flex_attn=True \
--pad=128`

**Comment by JeyesHan** [comment]
I think 4 GPUs is not enough for training 2B model from scratch with 5M data. I suggest you fine-tune the model.

**Comment by JeyesHan** [comment]
I think at least 16 GPUs (80GB vRAM).

**Comment by transform19** [comment]
Got it! I will give it a try. Thank you for your patience.

**Comment by RamseyZhu** [comment]
Could you let me know if you have successfully trained a usable model? If so, would you be willing to share your training configuration? I would be greatly grateful.
